src/test/groovy/InterfaceTest.groovy

```diff
@@ -91,4 +91,19 @@ final class InterfaceTest extends CompilableTestSupport {
             assert impl2.foo() == 'hello from Foo#foo'
         '''
     }
+
+    // GROOVY-11237
+    void testStaticInterfaceMethod() {
+        assertScript '''
+            interface Foo {
+                static hello(where) { "hello $where" }
+                static String BAR = 'bar'
+                String BAA = 'baa' // implicit static
+            }
+
+            assert Foo.hello('world') == 'hello world'
+            assert Foo.BAR == 'bar'
+            assert Foo.BAA == 'baa'
+        '''
```

Comment:
Can you also check that property methods are not generated:
```groovy
            groovy.test.GroovyAssert.shouldFail(MissingMethodException) {
                Foo.getBAR()
            }
            groovy.test.GroovyAssert.shouldFail(MissingMethodException) {
                Foo.getBAA()
            }
```

Reply:
Added